design/ceph/object/store.md

```diff
@@ -73,7 +75,53 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
   preservePoolsOnDelete: true
 ```
 
-If there is a `zone` section in object-store configuration, then the pool section in the ceph-object-zone resource will be used to define the pools.
+#### Pools shared by multiple CephObjectStore
+
+If user want to use existing pools for metadata and data, the pools must be created before the object store is created. This will be useful if multiple objectstore can share same pools. The detail of pools need to shared in `radosNamespaces` settings in object-store CRD. And for this option to work, `preservePoolsOnDelete` must be set to 'true'. Now the object stores can consume same pool isolated with different namespaces. User can create via [Pool CRD](/Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md), this is need to present before the object store is created.
+
+```yaml
+  radosNamespaces:
+    metadataPoolName: rgw-meta-pool
+    dataPoolName: rgw-data-pool
+  preservePoolsOnDelete: true
+```
+
+And the pool CRD will look like this:
+
+```yaml
```

Comment:
> Could you also add a section that shows what rgw configuration will be created for the rados namespaces and list which rados namespaces exactly will be created for each object store? This will help clarify [my question](https://github.com/rook/rook/pull/12140#discussion_r1179481908) about the `.rgw.root` pool which I'm still not clear about. If `.rgw.root` cannot be in a rados namespace, then we need to define its pool CR as well.

What about the `.rgw.root` pool?

Reply:
@travisn @BlaineEXE : The (.rgw.root)[https://docs.ceph.com/en/latest/radosgw/layout/#appendix-compendium] is common pool like `.mgr` and will shared by all the rgw servers in the cluster, so radosnamespace is not applicable for this pool IMO. If not specified it will be created with replica 1 settings. By looking through the code, for usual object stores, this will create metadata pool settings of the first object store. If the multisite settings are enabled for the first object store in the cluster then this will be created with replica setting 1, since Rook executes rados-admin commands for the realm, zonegroup which internally creates this pool.  

@cboldey : Do you recommend to create .rgw.root separately with certain configuration for Rook workloads?

Reply:
This is why Rook currently creates the .rgw.root pool with the settings in the object store CR, to ensure the expected replication and other settings. So we will need to have a CR for this pool as well

Reply:
@travisn I agree that it is better to include replication settings for the special pool `.rgw.root`, should set a default configuration for that pool rather than relying on the metadata pool setting for the first objectstore / first objectzone?

Reply:
We don't want to use a default configuration for any pool, we need to define the pool settings in a CR. So for this scenario it seems we should expect a CephBlockPool to be created for the `.rgw.root` pool, which would define its replication.

Reply:
@travisn : What about the `.mgr` pool ??

Reply:
I think I am confused about whether the `.rgw.root` pool **cannot** be namespaced or whether it is your opinion -- that is what IMO (**I**n **M**y **O**pinion) means -- that it **should not** be namespaced. Could you clarify?

It seems to me that it might be possible to put the data that would go into `.rgw.root` into the metadata pool to save on the number of CRUSH rules, but I'm not certain.